app/src/main/java/org/mascota/ui/view/diary/DiaryDetailWriteFragment.kt

```diff
@@ -0,0 +1,229 @@
+package org.mascota.ui.view.diary
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Build
+import android.provider.MediaStore
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.Log
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.annotation.RequiresApi
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import org.mascota.R
+import org.mascota.databinding.FragmentDiaryDetailWriteBinding
+import org.mascota.ui.base.BindingFragment
+import org.mascota.ui.view.diary.adpter.SpinnerAdapter
+import org.mascota.ui.view.diary.data.SpinnerModel
+import java.time.LocalDate
+import java.time.LocalDate.now
+import java.time.format.DateTimeFormatter
+
+
+@Suppress("DEPRECATION")
+class DiaryDetailWriteFragment : BindingFragment<FragmentDiaryDetailWriteBinding>(R.layout.fragment_diary_detail_write){
+
+    val REQ_STORAGE_PERMISSION: Int = 1
+    val REQ_GALLERY = 12
+    @RequiresApi(Build.VERSION_CODES.O)
+    val current_date : LocalDate = now()
+    @RequiresApi(Build.VERSION_CODES.O)
+    val formatter = DateTimeFormatter.ofPattern("yyyy.MM.dd")
+    @RequiresApi(Build.VERSION_CODES.O)
+    val tv_today = current_date.format(formatter).toString()
+
+    private lateinit var spinnerAdapter: SpinnerAdapter
+    private var item_spin = arrayListOf<SpinnerModel>()
+
+    val image = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
```

Comment:
해당 클래스에서 쓰는 변수의 경우 모두 private 한정자를 붙여주세요 :) 객체지향적으로도 올바른 방식(캡슐화의 관점에서)입니다 :)

Reply:
> val REQ_STORAGE_PERMISSION: Int = 1
    val REQ_GALLERY = 12

그리고 이런 상수의 경우 companion object안에

```suggestion
    companion object {
        const val REQ_STORAGE_PERMISSION: Int = 1
        const val REQ_GALLERY = 12
    }
```
이런식으로 선언 해주세요 :)